R/class.R

```diff
@@ -245,7 +245,8 @@ new_object <- function(.parent, ...) {
   args <- list(...)
   nms <- names(args)
 
-  object <- .parent %||% class_construct(class@parent)
+  # TODO: Some type checking on `.parent`?
+  object <- .parent
```

Comment:
Could maybe use something based on `check_can_inherit()`? But I think it's also fine to leave for this PR, since generally we done little to support custom constructors.